src/ConnectionPool.cpp

```diff
@@ -70,6 +70,29 @@ int ConnectionPool::init(const char* const * hosts, const uint32_t* ports, const
   return rv;
 }
 
+int ConnectionPool::update_servers(const char* const* hosts, const uint32_t* ports, const size_t n,
+                                   const char* const* aliases) {
+  int rv = 0;
+  if (m_nConns != n) {
+    return 1;
+  }
+  for (size_t i = 0; i < m_nConns; i++) {
+    if (m_conns[i].hasAlias() && (strcmp(m_conns[i].name(), aliases[i]) != 0)) {
+      return 2;
+    }
+  }
+  for (size_t i = 0; i < m_nConns; i++) {
+    if ((strcmp(m_conns[i].host(), hosts[i]) == 0) && (m_conns[i].port() == ports[i])) {
+      rv += -1;
+    } else {
+      m_conns[i].markDead("update servers", 1);
```

Comment:
1 is magic number, consider use  conn m_retryTimeout

Reply:
I would prefer a min int but not `retryTimeout` from configuration in this case.

Reply:
https://github.com/douban/libmc/pull/73/files#diff-4a3c6683d33b2ba8f3320fbf5e943cf3R196 this pr also add one situation that use markDead after quit。

so i think we need to make it more consistent。

should we pass reason？
should we pass reason not define as keywords?
the pr's addition consider is that it wont outout any log when quit? at lease not at the same logging level

Reply:
> I would prefer a min int rather than retryTimeout from configuration in this case.

@everpcpc i dont think it's a reason to use magic number。you can use 1/n of retryTimeout

Reply:
@everpcpc my key point, 1 is hard code number。if i dont like it, i cant change it easy。